src/mds/FSMap.h

```diff
@@ -472,6 +472,22 @@ class FSMap {
     fs.mds_map.modified = ceph_clock_now();
   }
 
+  void swap_fscids(fs_cluster_id_t fscid1, fs_cluster_id_t fscid2)
+  {
+    auto fs1 = filesystems.at(fscid1);
+    auto fs2 = filesystems.at(fscid2);
+    filesystems[fscid1] = fs2;
+    filesystems[fscid2] = fs1;
+    for (auto it : mds_roles) {
```

Comment:
This loop should no longer be necessary if the file system is failed (which you should verify in this method).

Reply:
Okay. This method needs to be re-worked. I'll take care of these details today when I do it.